src/OpenToolkit.Mathematics/Geometry/Box2.cs

```diff
@@ -63,23 +54,14 @@ public Vector2 Max
             {
                 if (value.X < _min.X)
                 {
-                    _max.X = _min.X;
                     _min.X = value.X;
                 }
-                else
-                {
-                    _max.X = value.X;
-                }
-
                 if (value.Y < _min.Y)
                 {
-                    _max.Y = _min.Y;
                     _min.Y = value.Y;
                 }
-                else
-                {
-                    _max.Y = value.Y;
-                }
+
+                _max = value;
```

Comment:
why? that changes the meaning?

If I have a Box e.g. Box2(0,0,5,5)
and set min to (0,10)
Box2 will be (0,10,5,10), therefore it having Height of zero.
I would expect it to be similar to the rubberband rectangle in e.g. paint. If you drag the minimal value over to the other side, it will automatically set the max value automatically...

Reply:
https://discordapp.com/channels/337627185248468993/525453014618865664/586531888881991681